src/JustSaying/AwsTools/MessageHandling/IamPolicyExtensions.cs

```diff
@@ -50,4 +49,43 @@ internal static string BuildPolicyJson(SnsPolicyDetails policyDetails)
     ]
 }}";
     }
+
+    internal static string BuildIamPolicyJson(this SqsPolicyDetails policyDetails)
+    {
+        var sid = Guid.NewGuid().ToString().Replace("-", "");
+
+        var resource = policyDetails.QueueArn;
+
+        var topicArnWildcard = string.IsNullOrWhiteSpace(policyDetails.SourceArn)
+            ? "*"
+            : CreateTopicArnWildcard(policyDetails.SourceArn);
+
+        var policyJson = $@"{{
+    ""Version"" : ""2012-10-17"",
+    ""Statement"" : [
+        {{
+            ""Sid"" : ""{sid}"",
+            ""Effect"" : ""Allow"",
+            ""Principal"" : {{
+                ""AWS"" : ""*""
+            }},
+            ""Action""    : ""sqs:SendMessage"",
+            ""Resource""  : ""{resource}"",
+            ""Condition"" : {{
+                ""ArnLike"" : {{
+                    ""aws:SourceArn"" : ""{topicArnWildcard}""
+                }}
+            }}
+        }}
+    ]
+}}";
```

Comment:
This could be updated to use the new raw string literals language feature, as we build using the .NET 7 sdk.

```suggestion
var policyJson = $$"""
{
"Version" : "2012-10-17",
"Statement" : [
    {
        "Sid" : "{{sid}}",
        "Effect" : "Allow",
        "Principal" : {
            "AWS" : "*"
        },
        "Action"    : "sqs:SendMessage",
        "Resource"  : "{{resource}}",
        "Condition" : {
            "ArnLike" : {
                "aws:SourceArn" : "{{topicArnWildcard}}"
            }
        }
    }
  ]
}
""";
```

Reply:
Ah interesting, I hadn't used that yet.